SolarEnergy/Aero/aero.py

```diff
@@ -0,0 +1,226 @@
+import math
+from scipy import optimize
+import CbData
+from waypoints import Waypoint
+import geopy
+import geopy.distance
+from geographiclib.geodesic import Geodesic
+import numpy
+
+gDEBUG = False
+
+class Aeroplane:
+    #Class set to quickly calc for heavier than air flight.
+    # Power = 0.5 * density * Vf**3 * S * C * + Weight**2 / ((0.5 *density*V*S) * 3.14159 * e*AR)
+
+    def __init__(self, wingspan=35.8, wingarea=174.0, panelarea=160.0, weight=2950.0, paradragcoef=0.025, OswaldEff=0.8):
+        # Fixed aerodynamic parameters. Imperial Units !!
+        self.b = wingspan # ft
+        self.S = wingarea # ft2, note this is the wing-area not the PV area.
+        self.PV = panelarea
+        self.CD0 = paradragcoef # parasitic (non-lift drag) coeficient
+        self.weight = weight # in lbs (US/UK the same)
+        self.e = OswaldEff   # This is a correction for excess drag (lift), compared to ideal elliptical wing.
+        self._SetAspectRatio()
+        # Positional stuff
+        self.speed = None # mph
+        self.altitude = None
+        self.chkspeed = None # When checking stall/max speed.
+        self.position = None # Position in Lat/Long/alt. Waypoint class.
+        self.oldpos = None # The position at start of iteration
+        self.time = 0.0 # Journey Timer, in minutes.
+        self.dt = 10.0  # minutes, clock increment.
+        # Engine related
+        self.maxpower = 70.0
+        self.maxbattpack = 633.0*260.0 * 3600.0   # BattWeight(kg)*Density, in Watt-hrs, conv. to Joules with 3600 (60x60)
+        self.currentbatt = self.maxbattpack # Start at full battery power
+        self.egain = 0.0 # Energy loss & gain in last time increment.
+        self.eloss = 0.0
+        self.irrad = 0.0 # Irradiance for iteration
+        self.record = []
+        self.recordPD = []
+        # Constants (propeff will depend on height - ie. airdensity, to a 1/3 power I think ?)
```

Comment:
Can we get rid of the 'I think?'

Reply:
Removed this comment, it depends on the propeller. a fixed 85% is going to be close enough for these purposes.